verifier/src/identity.rs

```diff
@@ -299,96 +328,212 @@ where
 /// identity, while the `TrustedIdentitiesVerifier` is for verifying one of *multiple* identities.
 #[derive(Debug, Clone)]
 enum TrustedIdentityVerifier {
-    MrEnclave(TrustedMrEnclaveIdentityVerifier),
-    MrSigner(TrustedMrSignerIdentityVerifier),
+    MrEnclave(And<MrEnclaveVerifier, AdvisoriesVerifier>),
+    MrSigner(And<MrSignerVerifier, AdvisoriesVerifier>),
```

Comment:
I dropped the dedicated structs and used the And types directly here, mainly to thin out the types, there are quite a few in this file and it's getting hard to follow.